zerver/lib/request.py

```diff
@@ -491,3 +491,28 @@ def _wrapped_req_func(
         return return_value
 
     return _wrapped_req_func
+
+
+def extract_stream_id(req_to: str) -> int:
+    # Recipient should only be a single stream ID.
+    try:
+        stream_id = int(req_to)
+    except ValueError:
+        raise JsonableError(_("Invalid data type for stream ID"))
+    return stream_id
+
+
+def extract_direct_message_recipient_ids(req_to: str) -> List[int]:
+    try:
+        user_ids = orjson.loads(req_to)
+    except orjson.JSONDecodeError:
+        user_ids = req_to
+
+    if not isinstance(user_ids, list):
+        raise JsonableError(_("Invalid data type for recipients"))
+
+    for user_id in user_ids:
+        if not isinstance(user_id, int):
+            raise JsonableError(_("Recipient list may only contain user IDs"))
+
+    return list(set(user_ids))
```

Comment:
Should we put these functions in a new file (These functions are used in `set typing status` & `scheduled messages` codepath) ? 

Among existing files this looked the best option to me as they are mainly used in `views` to extract request data.

Reply:
Yeah, this file is meant to be just for the `has_request_variables` system, which we're planning to delete.

I think putting these in a new `zerver/lib/recipient_parsing.py`, and then as a follow-up moving `extract_private_recipients` / `extract_stream_indicator` to that file, would be a decent strategy. And we can make a new `zerver/tests/test_recipient_parsing` for that new file as well.

Reply:
Made the changes -- 4th commit.